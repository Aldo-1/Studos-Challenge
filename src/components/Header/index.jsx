import useStyles from './styles'
import User from 'assets/images/profile_image.png'
import Toggle from 'components/Toggle'
import { useTheme } from 'react-jss'

const Header = ({ children }) => {
  const theme = useTheme()
  const classes = useStyles({ theme })
  return (
    <header className={classes.header}>
      <div className={classes.user}>
        <img src={User} alt="Foto do usuario" />
        <h2>Olá, João Paulo</h2>
      </div>
      <div className={classes.darkMode}>
        <span>Modo escuro</span>
        <Toggle />
      </div>
    </header>
  )
}

export default Header