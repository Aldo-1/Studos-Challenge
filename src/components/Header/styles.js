import { createUseStyles } from 'react-jss'

const useStyles = createUseStyles({
  header: {
    display: 'flex',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingTop: '2rem'
  },
  user: ({ theme }) => ({
    display: 'flex',
    alignItems: 'center',
    '& h2': {
      fontSize: theme.font.sizes.small,
      color: theme.colors.grape,
      marginLeft: theme.spacings.xsmall,
      maxWidth: '10rem',
      '@media (min-width: 767px)': {
        fontSize: theme.font.sizes.large,
        maxWidth: 'initial',
        marginLeft: theme.spacings.medium
      }
    }

  }),
  darkMode: ({ theme }) => ({
    display: 'flex',
    alignItems: 'center',
    '& > span': {
      fontSize: theme.font.sizes.small,
      color: theme.colors.grape,
      marginRight: theme.spacings.xxsmall,
      fontWeight: theme.font.bold,
      display: 'block',
      maxWidth: '5rem',
      '@media (min-width: 767px)': {
        marginRight: theme.spacings.small,
        maxWidth: 'initial'
      }
    }
  })
})

export default useStyles;