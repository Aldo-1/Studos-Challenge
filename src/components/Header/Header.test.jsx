import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helper'
import Header from './index'

describe('<Header />', () => {
  it('should render the component', () => {
    const { container } = renderWithTheme(<Header />)

    expect(screen.getByRole('heading', { name: /Olá, João Paulo/i })).toBeInTheDocument()
    expect(screen.getByText(/modo escuro/i)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})