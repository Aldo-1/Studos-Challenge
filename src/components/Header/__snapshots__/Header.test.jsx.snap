// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render the component 1`] = `
<header
  class="header-0-2-1"
>
  <div
    class="user-0-2-2 user-d0-0-2-4"
  >
    <img
      alt="Foto do usuario"
      src="profile_image.png"
    />
    <h2>
      Olá, João Paulo
    </h2>
  </div>
  <div
    class="darkMode-0-2-3 darkMode-d1-0-2-5"
  >
    <span>
      Modo escuro
    </span>
    <label
      class="switch-0-2-6"
    >
      <input
        class="inputCheck-0-2-8"
        type="checkbox"
        value=""
      />
      <span
        class="slider-0-2-7 slider-d0-0-2-10 round-0-2-9 round-d1-0-2-11"
      />
    </label>
  </div>
</header>
`;
